### ISSUE TITLE
`ThrowsException` constructor removed in 2.5.0

### ISSUE THREAD

**Issue by bgrainger** [xunit/xunit#2741]
The following code worked in xunit 2.4.2:

```csharp
// no exception was thrown
throw new ThrowsException(typeof(TException));

// wrong exception was thrown
throw new ThrowsException(typeof(TException), actualException);
```

The first method has a replacement in 2.5.0: https://github.com/xunit/assert.xunit/blob/c02d676a36e75da394c2f6ed9d54ffc54d3bf973/Sdk/Exceptions/ThrowsException.cs#L76

```csharp
throw ThrowsException.ForNoException(typeof(TException));
```

The second method could have a replacement, but it's `internal`, not `public`: https://github.com/xunit/assert.xunit/blob/c02d676a36e75da394c2f6ed9d54ffc54d3bf973/Sdk/Exceptions/ThrowsException.cs#L38

This was a change in https://github.com/xunit/assert.xunit/commit/eae257b617f348fe2ba1c68871c9d2b908c3be98.

It would be helpful if the `ThrowsException.ForIncorrectExceptionType` method was also made `public`.

(Additionally, the constructor has become private: https://github.com/xunit/assert.xunit/blob/c02d676a36e75da394c2f6ed9d54ffc54d3bf973/Sdk/Exceptions/ThrowsException.cs#L22. This would be a breaking change for anyone deriving from this non-`sealed` class.)

**Comment by bradwilson** [comment]
This is documented in [the release notes](https://xunit.net/releases/2.5.0):

> For the most part, developers should not notice much compile-time differences (aside from perhaps the occasional type ambiguity that may come from new overloads); there are also some breaking binary changes, most prominently with the gutting and reworking of the exception classes that are thrown by the individual assertions. Because of some of these breaking changes, we felt it was warranted to bump the version up to 2.5.0.

None of the exceptions other than the base `XunitException` have a public constructor any more, because the exceptions are now using a pattern where they are entirely responsible for generating their own messages.

> It would be helpful if the ThrowsException.ForIncorrectExceptionType method was also made public

The factory methods should all be public. That one must've been overlooked.

**Comment by bradwilson** [comment]
Public constructor fix is in `2.5.1-pre.2`.
